### DISCUSSION TITLE
Fallback for interpolation value

### DISCUSSION THREAD

**Discussion by fiznool** [i18next/i18next#1621] [Ideas]
Apologies if this has been asked elsewhere, I had a quick search but couldn't find anything relevant.

My use case: providing a fallback value for an interpolation, if the supplied value is undefined or null. For example, with the following translation string:

``` json
{
  "throttledCode": "A code was recently sent to {{ email }}."
}
```

When calling this as follows (e.g. from a route handler):

``` js
const suppliedEmailAddress = req.body.email;
t('throttledCode', { email: suppliedEmailAddress });
```

If the user did not supply an email address, or there was a bug introduced in parsing the email address from the body, I'd like to show the following: _A code was recently sent to this email address._

Currently, I'd need to do the following to achieve this:

``` json
{
  "throttledCode": "A code was recently sent to {{ email }}.",
  "throttledCodeNoEmailFallback": "this email address
}
```

``` js
const suppliedEmailAddress = req.body.email;
const email = suppliedEmailAddress ?? t('throttledCodeNoEmailFallback');
t('throttledCode', { email });
```

What would be really nice is some sort of syntactic sugar in the translation file to use a fallback in case the supplied value is undefined or null. For example:

``` json
{
  "throttledCode": "A code was recently sent to {{ email ?? 'this email address' }}."
}
```

Then, I wouldn't need to do anything when using the `t` function, i18next would figure out whether to show the supplied `email` value or the fallback.

Has this been discussed before? If not, would it be something to consider?

**Comment by jamuhl** [comment]
```js
{
  "throttledCode": "A code was recently sent to {{ email ?? 'this email address' }}."
}
```

is string concatenating - which is a bad practice in i18n -> as this works not for all languages...

so your solution with 

```js
const suppliedEmailAddress = req.body.email;
const email = suppliedEmailAddress ?? t('throttledCodeNoEmailFallback');
t('throttledCode', { email });
```
while less clean in code - is better in regard of being able to translate proper
